packages/fxa-auth-server/scripts/write-emails-to-disk.js

```diff
@@ -46,37 +46,38 @@ const mailSender = {
 
   close: function () {},
 };
-
-require('../lib/senders/translator')(
-  config.i18n.supportedLanguages,
-  config.i18n.defaultLanguage
-)
-  .then((translator) => {
-    return createSenders(
-      log,
-      config,
-      { check: () => Promise.resolve() },
-      translator,
-      {},
-      mailSender
-    );
-  })
-  .then((senders) => {
-    const mailer = senders.email._ungatedMailer;
-    checkMessageType(mailer);
-
-    ensureTargetDirectoryExists();
-
-    return sendMails(mailer, getMessageTypesToWrite(mailer));
-  })
-  .then(() => {
-    console.info('done');
-    process.exit(0);
-  })
-  .catch((err) => {
-    console.error(err.stack);
-    process.exit(1);
-  });
+if (require.main === module) {
```

Comment:
This is why the mocha process was exiting with a success exit code. It was executing this when the test file `require`ed it. Since it was the mocha process running it `process.exit(0)` was always terminating mocha with an ok exit code.